### PR TITLE
feat: esm migration - Adopt Pure ESM Build and Refresh Test Harness for Vite 7 and Vitest 3

### PR DESCRIPTION
**Pull request by hmrkx**
- Raise the SDK baseline to Node 20+ and publish a single bundled ESM entry (dist/index.mjs) with embargoed CJS exports
  - Upgrade runtime/tooling deps (ethereumjs, viem, vitest/vite, typescript, eslint, etc.) to their latest majors and align tsup/
  tsconfig with ES2022 bundler resolution
  - Harden lint rules against bare core-module imports and CommonJS fallbacks
  - Split test bootstrap (testEnvironment.ts) from reusable client-storage helpers, streamline them for Node 20, and ensure dotenv
  - global matchers load once via Vitest setup
  - validate the new stack with pnpm build, pnpm exec tsc --noEmit, and the full unit suite on Vite 7/Vitest 3

### CODE REVIEW

**Comment by yilmazbahadir** [comment]
### Test Plan
1. Run the [lattice-simulator](https://github.com/GridPlus/lattice-simulator):
`LATTICE_AUTO_APPROVE=true DEBUG_SIGNING=1 pnpm dev`

3. 
```
DEBUG_EVM_SIGN=1 baseUrl=http://127.0.0.1:3000 DEVICE_ID=SD0001 PASSWORD=12345678 PAIRING_SECRET=12345678 APP_NAME=lattice-manager pnpm e2e-sign-evm-tx -- --reporter=basic
```

 ✓ src/__test__/e2e/signing/evm-tx.test.ts (36 tests) 47955ms
   ✓ EVM Transaction Signing - Unified Test Suite > Legacy Transactions > Simple ETH transfer - Legacy (1/5)  1415ms
   ✓ EVM Transaction Signing - Unified Test Suite > Legacy Transactions > Contract interaction - Legacy (2/5)  2364ms
   ✓ EVM Transaction Signing - Unified Test Suite > Legacy Transactions > Zero value transaction - Legacy (3/5)  1166ms
   ✓ EVM Transaction Signing - Unified Test Suite > Legacy Transactions > High nonce transaction - Legacy (4/5)  1047ms
   ✓ EVM Transaction Signing - Unified Test Suite > Legacy Transactions > Polygon Legacy transaction (5/5)  1275ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-1559 Transactions (Fee Market) > Simple ETH transfer - EIP-1559 (1/6)  1142ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-1559 Transactions (Fee Market) > DAI transfer - EIP-1559 (2/6)  2080ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-1559 Transactions (Fee Market) > Uniswap V2 swap - EIP-1559 (3/6)  1488ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-1559 Transactions (Fee Market) > WETH deposit - EIP-1559 (4/6)  1419ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-1559 Transactions (Fee Market) > High priority fee - EIP-1559 (5/6)  1020ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-1559 Transactions (Fee Market) > Polygon EIP-1559 transaction (6/6)  994ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-2930 Transactions (Access Lists) > Simple transfer with access list - EIP-2930 (1/4)  1189ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-2930 Transactions (Access Lists) > Contract interaction with multiple access entries - EIP-2930 (2/4)  1831ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-2930 Transactions (Access Lists) > Empty access list - EIP-2930 (3/4)  1142ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-2930 Transactions (Access Lists) > Complex DeFi interaction - EIP-2930 (4/4)  1459ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-7702 Transactions (Account Abstraction) > Real mainnet EIP-7702 transaction - Simple authorization (1/3)  1269ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-7702 Transactions (Account Abstraction) > Real mainnet EIP-7702 transaction - Different authorization (2/3)  1462ms
   ✓ EVM Transaction Signing - Unified Test Suite > EIP-7702 Transactions (Account Abstraction) > Real mainnet EIP-7702 transaction - High value authorization (3/3)  1299ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Maximum nonce value - Legacy (1/18)  1053ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Maximum gas price - Legacy (2/18)  1084ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Minimal gas limit - EIP-1559 (3/18)  1047ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Large data payload - EIP-1559 (4/18)  1803ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Contract creation - Legacy (5/18)  1479ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Zero gas price - Legacy (pre-EIP-155) (6/18)  1128ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Alternative chain IDs - Polygon (7/18)  1012ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Alternative chain IDs - BSC (8/18)  1264ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Alternative chain IDs - Avalanche (9/18)  1094ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Large chain ID - Palm Network (10/18)  1179ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Unknown chain ID (11/18)  1221ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Minimal value - 1 wei (12/18)  1047ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Scientific notation value - 1e8 wei (13/18)  1273ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Maximum safe value (14/18)  1248ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Contract creation - EIP-1559 (15/18)  1888ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > Contract creation - EIP-2930 (16/18)  1481ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > EIP-2930 with mixed storage keys (17/18)  979ms
   ✓ EVM Transaction Signing - Unified Test Suite > Edge Cases & Boundary Conditions > EIP-1559 with access list (18/18)  1148ms

 Test Files  1 passed (1)
      Tests  36 passed (36)
   Start at  11:45:15
   Duration  49.80s (transform 246ms, setup 17ms, collect 1.13s, tests 47.95s, environment 0ms, prepare 49ms)

File: src/__test__/utils/runners.ts [hunk 1]
```diff
@@ -57,26 +55,22 @@ export async function runEvm(
...
-    req.data.payload = tx.getMessageToSign(false);
+  if (!bypassSetPayload) {
+    const msgToSign = tx.getMessageToSign();
+    // Legacy tx returns an array of RLP components, typed tx returns Uint8Array
+    req.data.payload = Array.isArray(msgToSign) ? RLP.encode(msgToSign) : msgToSign;
   }
   // Request signature and validate it
   await client.connect(getDeviceId());
-  const resp = await client.sign(req);
+  const resp: any = await client.sign(req);
```
Comment (netbonus) [inline src/__test__/utils/runners.ts]: this should have a type

File: vite.config.ts [hunk 2]
```diff
@@ -1,49 +1,18 @@
 import { defineConfig } from 'vitest/config';
-import dts from 'vite-plugin-dts';
-import { resolve } from 'node:path';
 
 export default defineConfig({
-  build: {
```
Comment (netbonus) [inline vite.config.ts:6]: is our build output the same as it was before? what's different?

File: src/__test__/utils/viemComparison.ts [hunk 3]
```diff
@@ -35,22 +27,7 @@ export const getFoundryAccount = () => {
...
-    43114: avalanche,
-  };
-
-  return createWalletClient({
-    account,
-    chain: chains[chainId] || mainnet,
-    transport: http(),
-  });
-};
+// (Wallet client helper removed as unused)
```
Comment (netbonus) [inline src/__test__/utils/viemComparison.ts]:
```suggestion

```

File: src/__test__/utils/viemComparison.ts [hunk 4]
```diff
@@ -1,15 +1,7 @@
...
-  parseTransaction,
-  serializeTransaction,
-  type Address,
-  type Hex,
-  type TransactionSerializable,
-} from 'viem';
+import { parseTransaction, serializeTransaction, type Address, type Hex, type TransactionSerializable } from 'viem';
 import { privateKeyToAccount } from 'viem/accounts';
-import { mainnet, polygon, bsc, avalanche } from 'viem/chains';
+// Chains & wallet client not needed currently; compare via account signing.
```
Comment (netbonus) [inline src/__test__/utils/viemComparison.ts]:
```suggestion
```

File: src/__test__/utils/setup.ts
Comment (netbonus) [inline src/__test__/utils/setup.ts:1]: why remove this and refactor here?

File: src/__test__/utils/runners.ts [hunk 5]
```diff
@@ -26,25 +27,22 @@ export async function runTestCase(
   return parsedRes;
 }
 
+type MaybeGenericData = { encodingType?: number; payload?: unknown };
+
 export async function runGeneric(request: SignRequestParams, client: Client) {
   const response = await client.sign(request);
-  // If no encoding type is specified we encode in hex or ascii
-  const encodingType = request.data.encodingType || null;
+  const data = request.data as unknown as MaybeGenericData;
```
Comment (netbonus) [inline src/__test__/utils/runners.ts]: we should just get the real type here, instead of casting

File: package.json
Comment (netbonus) [inline package.json:1]: spacing shouldn't be changed

File: src/__test__/e2e/non-exportable.test.ts
Comment (netbonus) [inline src/__test__/e2e/non-exportable.test.ts:1]: do we need all of these whitespace changes?

File: vite.config.ts (hunk 2 above)
Comment (hmrkx) [reply vite.config.ts:6]: The main differences would now be:
  - Build moved from Vite to tsup.
      - Old: Vite library build with Rollup options and vite-plugin-dts.
      - New: tsup-only build, ESM output, types emitted by tsup.
  - Vite config now only configures Vitest.
  - Package is ESM-only ("type": "module"), with exports pointing to the ESM bundle.

As for the outputs:
  - Old outputs
      - Formats: es + umd.
      - Files: gridplus-sdk.es.js and gridplus-sdk.umd.js (with name: 'GridPlusSDK' and UMD globals mappings).
      - DTS via vite-plugin-dts.
  - New outputs (tsup)
      - Format: esm only (Node 20 target), file: dist/index.mjs, sourcemaps on, DTS at dist/index.d.ts.
      - Externals: all dependencies and peerDependencies are excluded from the bundle (imported at runtime).
          - Old config only externalized a subset, now it externalizes everything in deps/peerDeps.

**Review by hmrkx — COMMENTED** [review]
👍  `[feat: add CommonJS build alongside ESM](https://github.com/GridPlus/gridplus-sdk/pull/614/commits/a45ae2f74a0ef8084a9f2fe1d0edd868b466e753)`

**Review by netbonus — APPROVED** [review]
